### PR TITLE
Turnin

### PR DESCRIPTION
**Pull request by doctorMcbob**

### CODE REVIEW

File: proper_paranthetics.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+def fn(s):
+    """
+    shows the state of parenthasies
+    if
+    """
+    s=list(s)
+    s.reverse()
```
Comment (Jakeand3rson) [inline proper_paranthetics.py:7]: I'm not really sure why you need to reverse it?

File: proper_paranthetics.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+def fn(s):
+    """
+    shows the state of parenthasies
+    if
+    """
+    s=list(s)
```
Comment (Jakeand3rson) [inline proper_paranthetics.py:6]: Why are you taking the string and turning it into a list?

File: proper_paranthetics.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    s.reverse()
+    n=0
+    while s:
+        t = s.pop()
+        if t == "(": 
+            n += 1
+        elif t == ")": 
+            n -= 1
+
+        if n < 0:
```
Comment (Jakeand3rson) [inline proper_paranthetics.py:16]: you have elif above, why not here?

Comment (Jakeand3rson) [inline proper_paranthetics.py:16]: I get why you are checking if its greater than zero though. Good job on dealing with the edge cases.

File: proper_paranthetics.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            return -1
+    if n:
+        return 1
+    else:
+        return 0
+
+def test_fn():
```
Comment (Jakeand3rson) [inline proper_paranthetics.py:23]: This needs to be in another file like the other assignments. There are too many cases to test with it just down here at the bottom.

File: proper_paranthetics.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    import random
```
Comment (Jakeand3rson) [inline proper_paranthetics.py:24]: Why are you importing random?

File: proper_paranthetics.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    if n:
+        return 1
+    else:
+        return 0
+
+def test_fn():
+    import random 
+    assert 1 == fn("((()()()()())()(()(")
+    assert 0 == fn("(())")
+    assert -1 == fn(")(")
```
Comment (Jakeand3rson) [inline proper_paranthetics.py:27]: Good start on testing. There are a lot of edge cases you can test to see if your program really works.

File: proper_paranthetics.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    if
+    """
+    s=list(s)
+    s.reverse()
+    n=0
+    while s:
+        t = s.pop()
+        if t == "(": 
+            n += 1
+        elif t == ")":
```
Comment (Jakeand3rson) [inline proper_paranthetics.py:13]: is elif what you want here?

File: proper_paranthetics.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
+def fn(s):
+    """
```
Comment (Jakeand3rson) [inline proper_paranthetics.py:2]: There are some linting errors here. Please have all the linting errors dealt with before you submit.
